lib/nodeJavaBridge.js

```diff
@@ -189,9 +194,21 @@ function isWritable(obj, prop) {
   return (Object.getOwnPropertyDescriptor(obj, prop) || {}).writable !== false;
 }
 
+var reservedPropertyNames = [ 'name', 'attributes', 'caller' ];
+
+function usableName(obj, name) {
+  // Node 0.10 and 0.12 have different behavior for whether function.name is declared writable or not.
+  // This test is constructed so that any unwritable property name OR the name 'name' will be altered,
+  // to give us consistent behavior for the one known problem case of properties named 'name'.
+  if (!isWritable(obj, name) || _.includes(reservedPropertyNames, name)) {
```

Comment:
I feel we should eliminate the redundancy.  There could be users out there who're forced to use existing libs on node v0.10.  I wouldn't want to break them with this.